File: backend/src/schemas/image.py
from pydantic import BaseModel, Field


class GenerateImageRequest(BaseModel):
    prompt: str = Field(description="图片描述，中英文均可")
    aspect_ratio: str = Field(default="1:1", description="宽高比 1:1 / 16:9 / 9:16")
    img_count: int = Field(default=1, ge=1, le=4, description="生成数量 1-4")
    chat_group_id: int = Field(default=0, description="对话分组ID")

File: frontend/src/api/image.js
import request from './request'

export function generateImage(data) {
  return request({
    url: '/image/generate',
    method: 'post',
    data: {
      prompt: data.prompt,
      aspect_ratio: data.aspect_ratio || '1:1',
      img_count: data.img_count || 1,
      chat_group_id: Number(data.chat_group_id || 0)
    }
  })
}

export function getImageTaskStatus(taskId) {
  return request.get(`/image/status/${taskId}`)
}

export function getGeneratedImages(params = {}) {
  return request.get('/image/list', { params })
}

export function getGeneratedImage(imageId) {
  return request.get(`/image/${imageId}`)
}

export function deleteGeneratedImage(imageId) {
  return request.delete(`/image/${imageId}`)
}

export function publishGeneratedImage(imageId, visibility = 'public') {
  return request.post(`/image/${imageId}/visibility`, { visibility })
}
